nsxt/resource_nsxt_policy_ipsec_vpn_dpd_profile.go

```diff
@@ -0,0 +1,198 @@
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"retry_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  10,
+			},
+		},
+	}
+}
+
+func resourceNsxtPolicyIPSecVpnDpdProfileExists(id string, connector *client.RestConnector, isGlobalManager bool) (bool, error) {
+	client := infra.NewIpsecVpnDpdProfilesClient(connector)
+	_, err := client.Get(id)
+	if err == nil {
+		return true, nil
+	}
+
+	if isNotFoundError(err) {
+		return false, nil
+	}
+
+	return false, logAPIError("Error retrieving resource", err)
+}
+
+func resourceNsxtPolicyIPSecVpnDpdProfileCreate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	// Initialize resource Id and verify this ID is not yet used
+	id, err := getOrGenerateID(d, m, resourceNsxtPolicyIPSecVpnDpdProfileExists)
+	if err != nil {
+		return err
+	}
+
+	displayName := d.Get("display_name").(string)
+	description := d.Get("description").(string)
+	tags := getPolicyTagsFromSchema(d)
+	dpdProbeInterval := int64(d.Get("dpd_probe_interval").(int))
+	dpdProbeMode := d.Get("dpd_probe_mode").(string)
+	enabled := d.Get("enabled").(bool)
+	retryCount := int64(d.Get("retry_count").(int))
+
+	obj := model.IPSecVpnDpdProfile{
+		DisplayName:      &displayName,
+		Description:      &description,
+		Tags:             tags,
+		DpdProbeInterval: &dpdProbeInterval,
+		DpdProbeMode:     &dpdProbeMode,
+		Enabled:          &enabled,
+		RetryCount:       &retryCount,
+	}
+
+	// Create the resource using PATCH
+	log.Printf("[INFO] Creating IPSecVpnDpdProfile with ID %s", id)
+	client := infra.NewIpsecVpnDpdProfilesClient(connector)
+	err = client.Patch(id, obj)
+	if err != nil {
+		return handleCreateError("IPSecVpnDpdProfile", id, err)
+	}
+
+	d.SetId(id)
+	d.Set("nsx_id", id)
+
+	return resourceNsxtPolicyIPSecVpnDpdProfileRead(d, m)
+}
+
+func resourceNsxtPolicyIPSecVpnDpdProfileRead(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining IPSecVpnDpdProfile ID")
+	}
+
+	client := infra.NewIpsecVpnDpdProfilesClient(connector)
+	obj, err := client.Get(id)
+	if err != nil {
+		return handleReadError(d, "IPSecVpnDpdProfile", id, err)
+	}
+
+	d.Set("display_name", obj.DisplayName)
+	d.Set("description", obj.Description)
+	setPolicyTagsInSchema(d, obj.Tags)
+	d.Set("nsx_id", id)
+	d.Set("path", obj.Path)
+	d.Set("revision", obj.Revision)
+	d.Set("dpd_probe_interval", obj.DpdProbeInterval)
+	d.Set("dpd_probe_mode", obj.DpdProbeMode)
+	d.Set("enabled", obj.Enabled)
+	d.Set("retry_count", obj.RetryCount)
+
+	return nil
+}
+
+func resourceNsxtPolicyIPSecVpnDpdProfileUpdate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining IPSecVpnDpdProfile ID")
+	}
+
+	// Read the rest of the configured parameters
+	description := d.Get("description").(string)
```

Comment:
Are these type casting needed? Seems calling Get on schema will return the correct data type
https://github.com/hashicorp/terraform/blob/v0.14.11/helper/schema/schema.go#L65

Reply:
`Get` returns `interface{}`, so unless we type cast here compiler would complain..